src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -360,14 +360,14 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
 
   protected def elemEquals(key1: A, key2: A): Boolean = (key1 == key2)
 
-  // Note:
-  // we take the most significant bits of the hashcode, not the lower ones
-  // this is of crucial importance when populating the table in parallel
-  protected final def index(hcode: Int) = {
+  /**
+    * Note: we take the most significant bits of the hashcode, not the lower ones
+    * this is of crucial importance when populating the table in parallel
+    */
+  protected final def index(hcode: Int): Int = {
     val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = Integer.numberOfLeadingZeros(ones)
```

Comment:
* @Ichoran, `numberOfLeadingZeros` call is optimized to `LZCNT` assembly call (see http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8045398):
```asm
0x00007fec011e69ba: lzcnt  %r8d,%edx          ;*invokestatic numberOfLeadingZeros
```
(remains an open question why `DeBruijn` is faster in the benchmarks, though)
* extracting it as a variable didn't improve the speed at all